### ISSUE TITLE
Allow re-ordering past queue items

### ISSUE THREAD

**Issue by dmint789** [OxygenCobalt/Auxio#223]
#### Describe the feature you want to implement:
Allow moving current and past songs in the queue. Moving the current song is essential if you want to continue listening from a different spot in the queue without having to drag **everything else** up instead.

#### Do other music players handle this? If so, how?
Yes, Simple Music Player and many other apps allow this and it's super convenient

#### Why do you think this will improve everyone's usage of Auxio?
It will be much easier to edit the current queue

#### Due Diligence:
- [v] I have read the [Contribution Guidelines](https://github.com/OxygenCobalt/Auxio/blob/dev/.github/CONTRIBUTING.md).
- [v] I have read the [Accepted Additions and Requests](https://github.com/OxygenCobalt/Auxio/blob/dev/info/ADDITIONS.md) document.
- [v] I have checked for this feature in the [FAQ](https://github.com/OxygenCobalt/Auxio/blob/dev/info/FAQ.md).

#### PS: Thank you so much for your work on this app and for your recent overhaul of the queue

**Comment by OxygenCobalt** [comment]
Implementing that is a technical nightmare with #110. I don't want to change queue behavior further right now until I can implement it.

**Comment by OxygenCobalt** [comment]
Re-opening since I finally have a prototype queue that should have the capability to handle this and gapless playback.

**Comment by dmint789** [comment]
Really glad to hear! This is genuinely the best music app for me, this was the last main feature I thought was lacking.

**Comment by OxygenCobalt** [comment]
Forgot to upload this when I added this due to some bugs I found, but here's a demo of the new queue editing functionality. 

https://user-images.githubusercontent.com/65370175/211449454-9da0e776-d6e0-4f3b-9b52-aca8b6b3861d.mp4

I not only managed re-ordering past items, but also allowing the user to move between past and future items, and even move/remove the current item itself. Should arrive in a 3.1.0 release.
